src/handlers/common.ts

```diff
@@ -18,30 +20,55 @@ const getStageFromFunctionName = (functionName: string) => {
 };
 
 /** Instantiate container using contextual information. */
-const createContainerFromEvent = (arg: uno.FunctionArg<uno.UnoEvent, Container>) => {
+const createContainerFromEvent = (
+  arg: uno.FunctionArg<uno.UnoEvent, Container>
+) => {
   /** True if running locally, under serverless-offline. */
   if (process.env.IS_OFFLINE || process.env.IS_LOCAL) {
     return createContainer({ mode: ExecutionMode.LocalDev, stage: "local" });
   }
 
   const mode = ExecutionMode.RestOfTheWorld;
 
-  const stage = arg.event.unoEventType === "http"
-    ? (arg.event as uno.HttpUnoEvent).original.requestContext.stage
-    : getStageFromFunctionName(arg.context.original.functionName);
+  const stage =
+    arg.event.unoEventType === "http"
+      ? (arg.event as uno.HttpUnoEvent).original.requestContext.stage
+      : getStageFromFunctionName(arg.context.original.functionName);
 
   return createContainer({
     mode,
     stage,
   });
 };
 
-export const httpFunc = () => uno.uno(unoAws.awsLambdaAdapter())
-  .use(uno.container(createContainerFromEvent))
-  .use(uno.cors())
-  .use(uno.defaultHttpMiddlewares())
-  .use(unoAws.principalFromRequestAuthorizer());
+const authMiddlware: uno.Middleware<uno.HttpUnoEvent, Container> = async function authMiddleware(
```

Comment:
With this you ensure every request is authenticated ?

Reply:
Yes, this middleware initialize the the auth service. The goal is to validate the token sent by the client in the "Authorization" header, decode it and make the decoded information available to the rest of the app via the authentication service.

Reply:
Note that of the request does not have a `Authorization` header, the middle ware does nothing. So it support _anonymous_ request too.